Networking/Networking/Mapper/ProductTagListMapper.swift

```diff
@@ -18,18 +18,38 @@ struct ProductTagListMapper: Mapper {
         decoder.userInfo = [
             .siteID: siteID
         ]
+        let hasDataEnvelope = response.hasDataEnvelope
 
         switch responseType {
-          case .load:
-            do {
+        case .load:
+            if hasDataEnvelope {
                 return try decoder.decode(ProductTagListEnvelope.self, from: response).tags
-            } catch {
+            } else {
                 return try decoder.decode([ProductTag].self, from: response)
             }
-          case .create:
-            return try decoder.decode(ProductTagListBatchCreateEnvelope.self, from: response).tags
-          case .delete:
-            return try decoder.decode(ProductTagListBatchDeleteEnvelope.self, from: response).tags
+        case .create:
+            let tags: [ProductTagFromBatchCreation] = {
+                if hasDataEnvelope {
+                    return (try? decoder.decode(ProductTagListBatchCreateEnvelope.self, from: response).data.tags) ?? []
+                } else {
+                    return (try? decoder.decode(ProductTagListBatchCreateContainer.self, from: response).tags) ?? []
+                }
+            }()
+            return tags
+                .filter { $0.error == nil }
+                .compactMap { (tagCreated) -> ProductTag? in
+                    if let name = tagCreated.name, let slug = tagCreated.slug {
+                        return ProductTag(siteID: tagCreated.siteID, tagID: tagCreated.tagID, name: name, slug: slug)
+                    }
+                    return nil
+                }
+
+        case .delete:
+            if hasDataEnvelope {
+                return (try? decoder.decode(ProductTagListBatchDeleteEnvelope.self, from: response).data.tags) ?? []
+            } else {
+                return (try? decoder.decode(ProductTagListBatchDeleteContainer.self, from: response).tags) ?? []
```

Comment:
❓ do we want to catch the errors for logging?

```suggestion
                return try decoder.decode(ProductTagListBatchDeleteEnvelope.self, from: response).data.tags
            } else {
                return try decoder.decode(ProductTagListBatchDeleteContainer.self, from: response).tags
```

Reply:
Also done in a66278f8d9cb907dd1b05fbfbba7c77996f1c5ff.